sklearn/utils/multiclass.py

```diff
@@ -350,6 +350,22 @@ def _check_partial_fit_first_call(clf, classes=None):
     return False
 
 
+def _validate_y_against_classes(y, classes):
```

Comment:
Is there a reason this shouldn't just be incorporated into `_check_partial_fit_first_call`?

And I suspect this should also handle multioutput classifiers, which basically means adding in a for-loop over the second dimension. Even without adding invariance tests that cover the multioutput case, you can test this validation function for that case.

Also, this function will behave very strangely if a sparse multilabel indicator matrix is passed in. Unless you have reason to believe this case should be supported here, I would just make this raises an appropriate error if a sparse matrix is passed in.